features/0685-pickup-v2/README.md

```diff
@@ -0,0 +1,186 @@
+Examples:
+
+```json=
+{
+    "@id": "123456781",
+    "@type": "https://didcomm.org/messagepickup/2.0/delivery-request",
+    "limit": 10,
+    "recipient_key": "<key for messages>"
+}
+```
+
+```json=
+{
+    "@type": "https://didcomm.org/messagepickup/2.0/delivery-request",
+    "limit": 1
+}
+```
+
+
+After receipt of this message, the mediator should deliver up to the limit indicated. 
+
+`recipient_key` is optional. When specified only return messages sent to that recipient key.
+
+If no messages are available to be sent, a status message MUST be sent immediately.
+
+Delivered messages MUST not be deleted until delivery is acknowledged by a `messages-received` message.
+
+
+### Message Received
+After receiving messages, the Recipient sends an ack message like this:
+```json=
+{
+    "@type": "https://didcomm.org/messagepickup/2.0/messages-received",
+    "message_tag_list": ["123","456"]
+}
+```
+
+`message_tag_list` identifiers are tags of each message received. The tag of each message is present in the encrypted form of the message as an artifact of encryption, and is indexed by the mediator. The tag is sufficiently unique within the scope of a recipient to uniquely identify the message.
+
+Upon receipt of this message, the `Mediator` knows that the messages have been received, and can remove them from the collection of queued messages with confidence. The mediator should send an updated `status` message reflecting the changes to the queue.
+
+### Multiple Recipients
+
+If a message arrives at a mediator addressed to multiple recipients, the message should be queued for each recipient independently. If one of the addressed recipients retrieves a message and indicates it has been received, that message must still be held and then removed by the other addressed recipients.
+
+## Live Mode
+Live mode is the practice of delivering newly arriving messages directly to a connected _Recipient_. It is disabled by default and only activated by the _Recipient_. Messages that arrive when Live Mode is off MUST be stored in the queue for retrieval as described above. If live mode is active, and the connection is broken, a new inbound connection starts with live mode off.
+
+Live mode must only be enabled when a persistent transport is used, such as websockets.
+
+_Recipients_ have three modes of possible operation for message delivery with various abilities and level of development complexity:
+
+1. Never activate live mode. Poll for new messages with a `status_request` message, and retrieve them when available.
+2. Retrieve all messages from queue, and then activate Live Mode. This simplifies message processing logic in the _Recipient_.
+3. Activate Live Mode immediately upon connecting to the _Mediator_. Retrieve messages from the queue as possible. When receiving a message delivered live, the Queue may be queried for any pending messages delivered to the same key for processing.
```

Comment:
Can you send a delivery request during live delivery? What is the behavior if the mediator has sent some messages to the recipient and then the recipient makes a delivery request--should the mediator track which messages have been sent but not received an ack for, or just respond with all messages and require the recipient to detect duplicate messages?